### DISCUSSION TITLE
Which commands causes indentation?

### DISCUSSION THREAD

**Discussion by stureandersen** [NilsSve/DfRefactor#48] [Ideas]
An idea would be to have a configurable list of commands that causes indentation when beautified..

I have a few custom commands that should cause indentation to happen, and it would bw handy if one could have DFRefactor respect these

**Comment by NilsSve** [comment]
This should be doable. The EditorNormalizeCase refactoring function already uses the "Scoped Words" and "KeyWords" found in the "Editor Settings" dialog.

Other functions uses these lists of keywords for other purposes. This was made to make the project as versatile and adjustable as possible. So you could e.g. add your "home brewed" command to the "Keyword" list and the Scintilla editor would then know about it.

The EditorReIndent function (or rather the Scintilla RefactorNormalizeCase method) would need to be rewritten, but it can be done.

**Comment by NilsSve** [comment]
It is worth mentioning that this is something for Wil van Antwerpen to have a look at as he knows all about the editor that is used by the project.. As he is very busy at the moment, a time frame can not be established for now.

**Comment by NilsSve** [comment]
Nope,  not so easy Wil told me.

It turned out that parts of the indentation logic are in the SciLexer32.dll and SciLexer64.dll files. So, I'm afraid it is hard-coded (for now). We'll see if this can be changed at a later stage.

**Comment by NilsSve** [comment]
Closed.
